src/basic/pidref.c

```diff
@@ -219,37 +219,32 @@ PidRef* pidref_free(PidRef *pidref) {
         return mfree(pidref);
 }
 
-int pidref_copy(const PidRef *pidref, PidRef *dest) {
-        _cleanup_close_ int dup_fd = -EBADF;
-        pid_t dup_pid = 0;
+int pidref_copy(const PidRef *pidref, PidRef *ret) {
+        _cleanup_(pidref_done) PidRef copy = PIDREF_NULL;
 
         /* If NULL is passed we'll generate a PidRef that refers to no process. This makes it easy to
          * copy pidref fields that might or might not reference a process yet. */
 
-        assert(dest);
+        assert(ret);
 
         if (pidref) {
                 if (pidref_is_remote(pidref)) /* Propagate remote flag */
-                        dup_fd = -EREMOTE;
+                        copy.fd = -EREMOTE;
                 else if (pidref->fd >= 0) {
-                        dup_fd = fcntl(pidref->fd, F_DUPFD_CLOEXEC, 3);
-                        if (dup_fd < 0) {
+                        copy.fd = fcntl(pidref->fd, F_DUPFD_CLOEXEC, 3);
+                        if (copy.fd < 0) {
                                 if (!ERRNO_IS_RESOURCE(errno))
                                         return -errno;
 
-                                dup_fd = -EBADF;
+                                copy.fd = -EBADF;
                         }
                 }
 
-                if (pidref->pid > 0)
-                        dup_pid = pidref->pid;
+                copy.pid = pidref->pid;
+                copy.fd_id = pidref->fd_id;
```

Comment:
If `pidref->fd >= 0` but we failed to copy the fd, should we copy the id??
-> #35948